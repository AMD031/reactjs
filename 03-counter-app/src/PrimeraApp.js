import React from 'react';
import PropTypes from 'prop-types';



const PrimeraApp = ({saludo, otra}) =>{
    return (
        <>
            <h1>{saludo}</h1>
             <p>{otra}</p>
        </>
        )
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
  };

  PrimeraApp.defaultProps = {
    otra: 'valor por defecto'
  };

export default PrimeraApp;