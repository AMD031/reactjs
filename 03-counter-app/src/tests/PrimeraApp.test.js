import React from 'react';
import {shallow} from 'enzyme';
//import {render} from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import '@testing-library/jest-dom';


describe('Pruebas <PrimeraApp/>', () => {


//     test('Debe mostrar el mensaje Hola ', () => {
//           const saludo = 'Hola';

//          // const wrapper = render(<PrimeraApp/> )
         
//          const {getByText} = render(<PrimeraApp saludo='Hola'}/>)

      
//          expect(getByText(saludo)).toBeInTheDocument();

//    })
    
        test('debe mostrar <PrimeraApp/> correctamente ', () => {
           
            const saludo = 'Hola';
            const wrapper = shallow(<PrimeraApp saludo={saludo} />);
            
            expect(wrapper).toMatchSnapshot();

        });
        

        test('debe mostrar el subtitulo enviado por props ', () => {
              
           const saludo ='Hola';
           const otra = 'texto de prueba';
           const wrapper = shallow(
           <PrimeraApp 
            saludo={saludo} 
            otra ={otra}
           />);


           const textoParrafo = wrapper.find('p').text() ;
           //console.log(textoParrafo);
           
           expect(textoParrafo).toBe(otra);

        })
        


})
