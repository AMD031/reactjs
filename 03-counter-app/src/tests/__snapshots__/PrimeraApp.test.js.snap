// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <PrimeraApp/> debe mostrar <PrimeraApp/> correctamente  1`] = `
<Fragment>
  <h1>
    Hola
  </h1>
  <p>
    valor por defecto
  </p>
</Fragment>
`;
